src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -110,9 +108,8 @@ public void createTopic(final EventType eventType) throws TopicCreationException
     }
 
     private long getTopicRetentionMs(final EventType eventType) {
-        return Optional.ofNullable(eventType.getOptions())
-                .map(EventTypeOptions::getRetentionTime)
-                .orElse(settings.getDefaultTopicRetentionMs());
+        final Long retentionTime = eventType.getOptions().getRetentionTime();
+        return retentionTime == null ? settings.getDefaultTopicRetentionMs() : retentionTime.longValue();
```

Comment:
Then I would say that it's callers responsibility to properly use the `create` method. It would better throw an exception than doing something by default. Allowing for silently misuse something could be harmful.